### PR TITLE
feat(l1): support snap capability message `GetTrieNodes`

### PR DESCRIPTION
**Pull request by fmoletta**
**Motivation**
Pass hive snap `GetTrieNodes` test

**Description**

* Add trie method `get_node`
* Add snap capability messages `GetTrieNodes` and `TrieNodes`
* Support processing `GetTrieNodes` message

Is part of #934

### CODE REVIEW

File: crates/storage/store/storage.rs [hunk 1]
```diff
@@ -811,6 +811,57 @@ impl Store {
         Ok(Some(proof))
     }
 
+    /// Receives the root of the state trie and a list of paths where the first path will correspond to a path in the state trie
+    /// (aka a hashed account address) and the following paths will be paths in the accoun's storage trie (aka hashed storage keys)
```
Comment (ElFantasma) [inline crates/storage/store/storage.rs:815]:
```suggestion
    /// (aka a hashed account address) and the following paths will be paths in the account's storage trie (aka hashed storage keys)
```

File: crates/storage/trie/nibble.rs [hunk 2]
```diff
@@ -322,7 +322,7 @@ impl NibbleVec {
         }
     }
 
-    #[cfg(test)]
+    //#[cfg(test)]
```
Comment (ElFantasma) [inline crates/storage/trie/nibble.rs:325]: Why not remove it instead of commenting it out?

File: crates/storage/store/storage.rs (hunk 1 above)
Comment (fmoletta) [reply crates/storage/store/storage.rs:815]: Thanks!

File: crates/storage/trie/nibble.rs (hunk 2 above)
Comment (fmoletta) [reply crates/storage/trie/nibble.rs:325]: I think this was leftover for when I was trying out stuff, I should restore it, ty for catching it!

**Review by Arkenan — APPROVED** [review]
LGTM

**Pull request merged**